tests/Microsoft.Identity.Web.Test.Integration/AcquireTokenForAppIntegrationTests.cs

```diff
@@ -0,0 +1,138 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Microsoft.Extensions.Caching.Memory;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Microsoft.Identity.Client;
+using Microsoft.Identity.Web.Test.Common;
+using Microsoft.Identity.Web.Test.Common.Mocks;
+using Microsoft.Identity.Web.Test.Common.TestHelpers;
+using Microsoft.Identity.Web.Test.LabInfrastructure;
+using Microsoft.Identity.Web.TokenCacheProviders.InMemory;
+using System;
+using System.Threading.Tasks;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.Identity.Web.Test.Integration
+{
+    public class AcquireTokenForAppIntegrationTests
+    {
+        TokenAcquisition _tokenAcquisition;
+        ServiceProvider _provider;
+        private MsalTestTokenCacheProvider _msalTestTokenCacheProvider;
+
+        private KeyVaultSecretsProvider _keyVault;
+        private string _ccaSecret;
+        private readonly ITestOutputHelper _output;
+
+        public AcquireTokenForAppIntegrationTests(ITestOutputHelper output) //test set-up
```

Comment:
Have a look at NUnit's [SetUp] and [OneTimeSetup] attributes. Some of the setup should be performed once, and some of it for each test: 

AcquireTokenForAppIntegrationTests -> once - I guess ctor is ok, but try to use [OneTimeSetup] as this is the recommended pattern. https://github.com/nunit/docs/wiki/SetUp-and-TearDown
InitializeTokenAcquisitionObjects -> once per test ([SetUp])

Reply:
i did look at that, but since we are already using xunit, i just stayed with their recommendations. we can change things over though.

---
In reply to: [385617575](https://github.com/AzureAD/microsoft-identity-web/pull/39#discussion_r385617575) [](ancestors = 385617575)

Reply:
XUnit allows the same behavior. I'll take care of it.

Reply:
thanks @pmaytak 

---
In reply to: [385919792](https://github.com/AzureAD/microsoft-identity-web/pull/39#discussion_r385919792) [](ancestors = 385919792)

Reply:
Yes, appologies, I mistakenly thought you were using NUnit instead of XUnit.